rskj-core/src/main/java/co/rsk/rpc/netty/rest/RestServerHandler.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2018 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.rpc.netty.rest;
+
+import co.rsk.rpc.netty.rest.dto.RestModuleConfigDTO;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RestServerHandler extends SimpleChannelInboundHandler<HttpObject> {
+    private static final Logger logger = LoggerFactory.getLogger(RestServerHandler.class);
+
+    private final RestServerDispatcher dispatcher;
+
+    public RestServerHandler(RestModuleConfigDTO restModuleConfigDTO) {
+        this.dispatcher = new RestServerDispatcher(restModuleConfigDTO);
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws Exception {
+
+        if (msg instanceof HttpRequest) {
+            logger.info("Client address: {}", ctx.channel().remoteAddress());
+
+            HttpRequest httpRequest = (HttpRequest) msg;
+
+            DefaultFullHttpResponse response = dispatcher.dispatch(httpRequest);
```

Comment:
Now that I think of it... what do you think about replacing `RestServerDispatcher` class by a `RestModules` enum? Since `RestModule` is an interface we could do something like this:

```
HttpRequest httpRequest = (HttpRequest) msg;
String uri = new URI(httpRequest.getUri()).getPath();

RestModules restModule = RestModules.lookup(uri, restModuleConfigDTO);

DefaultFullHttpResponse response;
if (restModule == null) { // uri not found or module disabled
    response = 404;
} else {
    response = restModule.processRequest(uri, httpRequest.getMethod());
}

if (response == null) {
    logger.info("Omitted response.");
    return;
}

ctx.writeAndFlush(response);
```

where `RestModules` is

```
public enum RestModules {

    HEALTHCHECK(new HealthCheckModule(), "/health-check", RestModuleConfigDTO::isHealthCheckModuleEnabled);

    private final RestModule restModule;

    private final String uri;

    private final Function<RestModuleConfigDTO, Boolean> isActive;

    RestModules(RestModule restModule, String uri, Function<RestModuleConfigDTO, Boolean> isActive) {
        this.restModule = restModule;
        this.uri = uri;
        this.isActive = isActive;
    }

    public static RestModules lookup(String moduleUri, RestModuleConfigDTO restModuleConfig) {
        for (RestModules module : RestModules.values()) {
            if (moduleUri.startsWith(module.uri) && module.isActive.apply(restModuleConfig)) {
                return module;
            }
        }
        return null;
    }

    public DefaultFullHttpResponse processRequest(String uri, HttpMethod method) {
        return restModule.processRequest(uri, method);
    }
}
```

WDYT?

Reply:
Did a couple of changes to use some of the suggestions, I think it ended up being nice.

Reply:
Well, TBH I don't see the need of having both `RestModuleLoader` and `RestServerDispatcher`. I think they share the same responsibility and are small enough to have just one class. Having both makes the code more complex to follow IMO. 

What benefit you see on having both?

I think module lookup and activation check could be moved to `RestModuleLoader` on a method like `public static RestModule getRestModule(String uri)`. Then, in `co.rsk.rpc.netty.rest.RestServerHandler#channelRead0`, you would simply:
- call `restModule.processRequest` if module found and active
- create the NOT FOUND response if module not found or inactive

WDYT?